src/records.rs

```diff
@@ -523,26 +543,50 @@ impl RecordBatchDecoder {
     {
         let mut records = Vec::new();
         while buf.has_remaining() {
-            Self::decode_batch(buf, &mut records, decompressor.as_ref())?;
+            let _ = Self::decode_into_vec(buf, &mut records, decompressor.as_ref())?;
         }
         Ok(records)
     }
-    fn decode_batch<B: ByteBuf, F>(
+
+    /// Decode the provided buffer into a vec of RecordSets.
+    pub fn decode_batches<B: ByteBuf>(buf: &mut B) -> Result<Vec<RecordSet>> {
```

Comment:
Hmm, I'm wondering if we should replace `fn decode` with `fn decode_batches`.

Reply:
Actually, what I'm thinking is we should have a single public decode function like:
```rust
pub fn decode<B: ByteBuf>(buf: &mut B) -> Result<RecordSet>
```

This way the user can call it as many times as needed to decode records, without the allocation of the outer Vec.

Reply:
Yeah that's nicer, as long as it's fine to break the API

Reply:
Yep, breaking API is fine, we will just release as a breaking version.
We should add a changelog entry for it though

Reply:
Yeah I will accept a `decode_batches`/`decode_all`.

I think this whole API should be refactored to use iterators so we can avoid allocating all these vectors for every batch. But that is definitely out of scope of this PR.

In the meantime, I agree lets solve the footgun.

Reply:
But the new function will need a basic test or two.

Reply:
do you think it's enough to update the existing tests to also check expected compression/version?

Reply:
@rukai @tychedelia what do you think about this interface/the tests?

Reply:
Thanks for the reminder.

> But the new function will need a basic test or two.

The new function `decode_all` needs at least one test.

Reply:
Indeed, I've switch some test cases into using `decode_all`

Reply:
@rukai sorry to bug you, wdyt?